lib/rspec/rails/extensions/active_support/json.rb

```diff
@@ -0,0 +1,63 @@
+if defined?(::RSpec::Mocks)
+  # Need to require the base file to setup internal support require helpers
+  require 'rspec/mocks'
+  require 'rspec/mocks/test_double'
+  ::RSpec::Mocks::TestDouble.module_exec do
+    # @private
+    def as_json(*args, &block)
+      return method_missing(:as_json, *args, &block) unless null_object?
+      __mock_proxy.record_message_received(:as_json, *args, &block)
+      nil
+    end
+
+    # @private
+    def to_json(*args, &block)
+      return method_missing(:to_json, *args, &block) unless null_object?
+      __mock_proxy.record_message_received(:to_json, *args, &block)
+      "null"
+    end
+
+    # @private
+    remove_method(:respond_to?)
```

Comment:
I'm concerned that if/when we change the `respond_to?` logic in rspec-mocks, it'll get ignored here in rspec-rails because you remove and redefine it.